File: elixir/bob/lib/bob.ex
defmodule Bob do
  def hey(input) do
    sanitized = String.trim(input)
    shouting = shouting?(sanitized)
    question = question?(sanitized)
    alpha = alpha?(sanitized)

    cond do
      silence?(sanitized) -> "Fine. Be that way!"
      alpha and shouting and question -> "Calm down, I know what I'm doing!"
      alpha and shouting -> "Whoa, chill out!"
      question -> "Sure."
      true -> "Whatever."
    end
  end

  defp alpha?(input), do: Regex.match?(~r/\p{L}+/, input)

  defp silence?(input), do: input == ""

  defp shouting?(input), do: String.upcase(input) == input

  defp question?(input), do: String.last(input) == "?"
end

File: javascript/bob/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const hey = (message) => {
  const trimed = message.trim();
  const isAlpha = trimed.match(/([a-z]|[A-Z])/)
  const isQuestion = trimed[trimed.length - 1] == "?"
  const isShouting = trimed.toUpperCase() == trimed

  if (isAlpha && isShouting && isQuestion) return "Calm down, I know what I'm doing!"
  if (isAlpha && isShouting) return "Whoa, chill out!"
  if (isQuestion) return "Sure."
  if (trimed == "") return "Fine. Be that way!"
  return "Whatever."
};
